### PR TITLE
update(validator-extras): remove regex `isEmpty` and add locale filter `isDecimal`

### PR DESCRIPTION
**Pull request by fncolon**
### Pull Request Checklist

_Please make sure to review and check all of these items:_

- [ ] Have you added new tests to prevent regressions?
- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

Depends on current version of validatorjs instead.

https://github.com/validatorjs/validator.js/blob/master/src/lib/isEmpty.js
https://github.com/validatorjs/validator.js/blob/master/src/lib/isDecimal.js

### Todos

- [x] remove `isEmpty` regex with changing to validatorjs `isEmpty`
- [x] add locale filter `isDecimal` with changing to validatorjs `isDecimal`

### CODE REVIEW

File: lib/utils/validator-extras.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ const extensions = {
     return !this.regex(str, pattern, modifiers);
   },
   isDecimal(str) {
-    return str !== '' && !!str.match(/^(?:-?(?:[0-9]+))?(?:\.[0-9]*)?(?:[eE][+-]?(?:[0-9]+))?$/);
```
Comment (sdepold) [inline lib/utils/validator-extras.js:42]: Uhm. Aren't we calling the exact same function here and create an endless loop?

Comment (sdepold) [reply lib/utils/validator-extras.js:42]: Could it be that you wanted to remove the function altogether?

Comment (fncolon) [reply lib/utils/validator-extras.js:42]: > Could it be that you wanted to remove the function altogether?

Hello, I'm still working on this PR sorry for late response.

Comment (fncolon) [reply lib/utils/validator-extras.js:42]: I will close this issue for now and will lookup after all new TODOS soon.

**Pull request closed without merging**